content/ADR-189-social-service-M2.md

```diff
@@ -0,0 +1,165 @@
+    Social service ->> User 1: Friend event (ACCEPT, DELETE, etc.)
+    end
+    User 1 ->> Social service: Disconnect WebSocket
+    Social service ->>- User 1: Disconnect WebSocket
+```
+
+## Implementation
+
+### Unity Renderer changes
+
+Given that Browser-interface is going to be slowly migrated into unity's codebase, the implementation of this feature will be done directly on unity's side too. Since this feature includes friends and friend requests, all of the code needed to handle friends and friendship requests will also be moved to unity, decoupling the dependency on Browser-interface. Still, channels, presence, and messaging will stay on the Browser interface side, but within time they will also be migrated to unity and will be able to leverage the new communication system.
+
+To prevent migrating all the code into unity-renderer's code and adding a lot of logic, this ADR proposes to add a new library in C# that will handle the logic around friendships and modify the logic in unity renderer to become more of an orchestrator of events.
+
+An example of a difference between the current implementation and the proposal is the following:
+
+#### Current:
+
+```mermaid
+sequenceDiagram
+    participant Renderer
+    participant Browser interface
+    participant Matrix client
+
+    Renderer ->>+ Browser interface: Get friends
+    Browser interface ->> Matrix client: Get friends
+    Matrix client -->> Browser interface: friends ids []
+    Browser interface ->> Browser interface: Calculate missing profiles
+    Browser interface ->> Renderer: Add profiles to catalog(Profiles [])
+    Browser interface -->>- Renderer: Friends []
+```
+
+#### Proposal:
+
+```mermaid
+sequenceDiagram
+    participant Renderer
+    participant Browser interface
+
+    Renderer ->>+ C# Social client: Get friends
+    C# Social client -->>- Renderer: Friends list
+    Renderer ->>+ Browser interface: Get missing profiles
+    Browser interface -->>- Renderer: Profiles[]
+    Renderer ->> Renderer: Store profiles in catalog
+```
+
+In this example, there are two main responsibilities of the flow:
+
+1. Obtaining the friendships list
+1. Storing the profiles in the catalog (to be able to render those friends)
+
+It can be seen that in the first example it's the browser interface's responsibility to know what profiles are missing and then send them to unity, making the browser interface the information orchestrator, in this new approach, Renderer is the orchestrator.
+
+### Browser interface changes
+
+This new structure will also imply changes on the browser interface side. First, all of the friendship logic will be deleted, maintaining only the necessary state so that the channels and messaging code can still function without any bugs. What will be deleted are the interactions with Unity around storing the friendships, friendship requests, and the logic for fetching profile information before sending the friends.
```

Comment:
```suggestion
This new structure will also require changes on the Browser Interface side. First, all of the friendship logic will be deleted, except for the necessary state to ensure that channels and messaging code continue to function without any issues. This will involve deleting the interactions with Unity around storing the friendships, friendship requests, and the logic for fetching profile information before sending the friends.
```